packages/kit/src/exports/vite/dev/index.js

```diff
@@ -408,7 +401,7 @@ export async function dev(vite, vite_config, svelte_config, illegal_imports) {
 							base: svelte_config.kit.paths.base,
 							assets
 						},
-						public_env: env.public,
+						public_env: {},
```

Comment:
I think this will set `env` to `{}` on the client during dev? Might need to set it to `null` (as in 'don't call `set_public_env` inside `start`')

Reply:
I'm looking into this now. During my testing, I triggered client-side navigations using `<a>` tags, and it still worked. Though I'm not sure why...

Reply:
Okay, maybe you can help me understand _why_ it still works without changing this. I even added to the existing test to make sure that clicknav-ing to the page results in a correct `env` value.

Reply:
Oh wait, I figured it out.

In non-dev-mode, we're just re-exporting the runtime `env` module. When we're running in dev, `$env/dynamic/*` is completely disconnected from the module it would normally re-export, so calling `set_*_env` _does_ set the value in `env-{public|private}.js` to `{}`, but that doesn't matter, because `$env/dynamic/*` isn't connected to it anyway. Long story short, we good.